File: csharp/Agentic.Commerce.Protocol/Models/FulfillmentOptionDigital.cs
// this file is @generated
using System.Text;
using Newtonsoft.Json;

namespace Agentic.Commerce.Protocol
{
    public class FulfillmentOptionDigital
    {
        [JsonProperty("id", Required = Required.Always)]
        public required string Id { get; set; }

        [JsonProperty("title", Required = Required.Always)]
        public required string Title { get; set; }

        [JsonProperty("subtitle")]
        public string? Subtitle { get; set; } = null;

        [JsonProperty("subtotal", Required = Required.Always)]
        public required long Subtotal { get; set; }

        [JsonProperty("tax", Required = Required.Always)]
        public required long Tax { get; set; }

        [JsonProperty("total", Required = Required.Always)]
        public required long Total { get; set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("class FulfillmentOptionDigital {\n");
            sb.Append("  Id: ").Append(Id).Append('\n');
            sb.Append("  Title: ").Append(Title).Append('\n');
            sb.Append("  Subtitle: ").Append(Subtitle).Append('\n');
            sb.Append("  Subtotal: ").Append(Subtotal).Append('\n');
            sb.Append("  Tax: ").Append(Tax).Append('\n');
            sb.Append("  Total: ").Append(Total).Append('\n');
            sb.Append("}\n");
            return sb.ToString();
        }
    }
}

File: ruby/lib/agentic_commerce_protocol/models/fulfillment_option_digital.rb
# frozen_string_literal: true
# This file is @generated
require "json"

module AgenticCommerceProtocol
  class FulfillmentOptionDigital
    attr_accessor :id
    attr_accessor :title
    attr_accessor :subtitle
    attr_accessor :subtotal
    attr_accessor :tax
    attr_accessor :total

    ALL_FIELD ||= ["id", "title", "subtitle", "subtotal", "tax", "total"].freeze
    private_constant :ALL_FIELD

    def initialize(attributes = {})
      unless attributes.is_a?(Hash)
        fail(
          ArgumentError,
          "The input argument (attributes) must be a hash in `AgenticCommerceProtocol::FulfillmentOptionDigital` new method"
        )
      end

      attributes.each do |k, v|
        unless ALL_FIELD.include?(k.to_s)
          fail(ArgumentError, "The field #{k} is not part of AgenticCommerceProtocol::FulfillmentOptionDigital")
        end

        instance_variable_set("@#{k}", v)
        instance_variable_set("@__#{k}_is_defined", true)
      end
    end

    def self.deserialize(attributes = {})
      attributes = attributes.transform_keys(&:to_s)
      attrs = Hash.new
      attrs["id"] = attributes["id"]
      attrs["title"] = attributes["title"]
      attrs["subtitle"] = attributes["subtitle"]
      attrs["subtotal"] = attributes["subtotal"]
      attrs["tax"] = attributes["tax"]
      attrs["total"] = attributes["total"]
      new(attrs)
    end

    def serialize
      out = Hash.new
      out["id"] = AgenticCommerceProtocol::serialize_primitive(@id) if @id
      out["title"] = AgenticCommerceProtocol::serialize_primitive(@title) if @title
      out["subtitle"] = AgenticCommerceProtocol::serialize_primitive(@subtitle) if @subtitle
      out["subtotal"] = AgenticCommerceProtocol::serialize_primitive(@subtotal) if @subtotal
      out["tax"] = AgenticCommerceProtocol::serialize_primitive(@tax) if @tax
      out["total"] = AgenticCommerceProtocol::serialize_primitive(@total) if @total
      out
    end

    # Serializes the object to a json string
    # @return String
    def to_json
      JSON.dump(serialize)
    end
  end
end
